vpr/src/route/connection_router.cpp

```diff
@@ -649,27 +755,47 @@ void ConnectionRouter<Heap>::evaluate_timing_driven_node_costs(t_heap* to,
 
     float total_cost = 0.;
     const t_conn_delay_budget* delay_budget = cost_params.delay_budget;
-    if (delay_budget) {
+
+    if (run_rcv && to->path_data != nullptr) {
+        to->path_data->backward_delay += cost_params.criticality * Tdel;
```

Comment:
Do we have data on whether we should just have the heap always keep track of split delay/cong and avoid specializing this for RCV?  If the cost for tracking delay/cong separately is cheap enough, I'd strongly prefer always keeping them seperate, even if on non-RCV routing we only end up caring about the cost?

Reply:
I could definitely give this a try. I don't see why it would have that significant a performance impact, but it's worth getting more data. This is a relatively easy refactor that could cut down some code.